src/topology/continuous_on.lean

```diff
@@ -325,17 +358,37 @@ end
 lemma continuous.comp_continuous_on {g : β → γ} {f : α → β} {s : set α}
   (hg : continuous g) (hf : continuous_on f s) :
   continuous_on (g ∘ f) s :=
-hg.continuous_on.comp hf (subset_univ _)
+hg.continuous_on.comp hf subset_preimage_univ
 
 lemma continuous_within_at.preimage_mem_nhds_within {f : α → β} {x : α} {s : set α} {t : set β}
   (h : continuous_within_at f s x) (ht : t ∈ nhds (f x)) : f ⁻¹' t ∈ nhds_within x s :=
 h ht
 
+lemma continuous_within_at.preimage_mem_nhds_within' {f : α → β} {x : α} {s : set α} {t : set β}
+  (h : continuous_within_at f s x) (ht : t ∈ nhds_within (f x) (f '' s)) :
+  f ⁻¹' t ∈ nhds_within x s :=
+begin
+  rw mem_nhds_within at ht,
+  rcases ht with ⟨u, u_open, fxu, hu⟩,
+  have : f ⁻¹' u ∩ s ∈ nhds_within x s :=
+    filter.inter_mem_sets (h (mem_nhds_sets u_open fxu)) self_mem_nhds_within,
+  apply mem_sets_of_superset this,
```

Comment:
Do you know about the `filter_upwards` tactic? In that case it would change much (`filter_upwards [this],` only saves a couple of keystrokes) but it can be useful to know.

Reply:
I try to avoid the use of fancy tactics when they are not clearly an improvement, as I think it is easier for a reader to check the effect of a lemma than the effect of a tactic. -- probably my Isabelle origin speaking, where you only use `auto` or `simp` and where I don't know how to do a rewrite since I never needed it! But thanks for the info on `filter_upwards` (that I had seen, but never used).